### ISSUE TITLE
Flask Debug Toolbar Does Not Check For A Compressed Response 

### ISSUE THREAD

**Issue by JackStouffer** [flask-debugtoolbar/flask-debugtoolbar#83]
If you have an extension like Flask Compress compressing the contents of your responses with gzip before it's sent back to the user, flask debug toolbar will throw an error because it assumes the response is UTF-8 HTML.

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/lib/python2.7/site-packages/flask_restful/__init__.py", line 270, in error_router
    return original_handler(e)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1479, in full_dispatch_request
    response = self.process_response(response)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1691, in process_response
    response = handler(response)
  File "/usr/local/lib/python2.7/site-packages/flask_debugtoolbar/__init__.py", line 202, in process_response
    response_html = response.data.decode(response.charset)
  File "/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x8b in position 1: invalid start byte
```

**Comment by mgood** [comment]
Can you try flipping the order that you add the extensions to your app? This should make the compression step happen after applying the toolbar.

**Comment by ffoxin** [comment]
Same problem with either Flask-Compress and manual gzip (recipe from http://flask.pocoo.org/snippets/122/). How can I reorder applying it?

**Comment by mgood** [comment]
Swapping the lines of code where you add the toolbar and add the compress extension should swap the order that they're executed. I think the "after_request" functions are processed in reverse order, so Flask-Compress should be the first extension initialized so that it's the last one called.

**Comment by jeffwidman** [comment]
Actually, I think we could throw a more useful error message here...

**Comment by jeffwidman** [comment]
Fixed in #154.
